packages/core/types/index.js

```diff
@@ -148,6 +148,8 @@ export type TargetDescriptor = {|
   +distEntry?: FilePath,
 |};
 
+export type SourceType = 'script' | 'module';
```

Comment:
It doesn't change anything effectively, but I think it would makes if this were
```suggestion
export type SourceType = null | 'script' | 'module';
```
instead. Because there really are 3 types and not 2?

Reply:
Yeah, this should address one of my concerns above. However, in `createEnvironment` it's currently not possible to explicitly set it to null.